tests/src/core/testqgsvectordataprovider.cpp

```diff
@@ -224,6 +242,75 @@ void TestQgsVectorDataProvider::featureAtId()
   QVERIFY( !feature.isValid() );
 }
 
+void TestQgsVectorDataProvider::sourceExtent()
+{
+  // 2d data
+  QgsVectorDataProvider *prLines = vlayerLines->dataProvider();
+
+  QGSCOMPARENEAR( prLines->sourceExtent().xMinimum(), -117.623, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().xMaximum(), -82.3226, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().yMinimum(), 23.2082, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().yMaximum(), 46.1829, 0.001 );
+
+  QGSCOMPARENEAR( prLines->sourceExtent3D().xMinimum(), -117.623, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().xMaximum(), -82.3226, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().yMinimum(), 23.2082, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().yMaximum(), 46.1829, 0.001 );
+  QVERIFY( std::isnan( prLines->sourceExtent3D().zMinimum() ) );
+  QVERIFY( std::isnan( prLines->sourceExtent3D().zMaximum() ) );
+
+
+  QgsVectorDataProvider *prPoints = vlayerPoints->dataProvider();
+
+  QGSCOMPARENEAR( prPoints->sourceExtent().xMinimum(), -118.889, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().xMaximum(), -83.3333, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().yMinimum(), 22.8002, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().yMaximum(), 46.872, 0.001 );
+
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().xMinimum(), -118.889, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().xMaximum(), -83.3333, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().yMinimum(), 22.8002, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().yMaximum(), 46.872, 0.001 );
+  QVERIFY( std::isnan( prPoints->sourceExtent3D().zMinimum() ) );
+  QVERIFY( std::isnan( prPoints->sourceExtent3D().zMaximum() ) );
+
+  // 3d data
+  QgsVectorDataProvider *prLines3D = vlayerLines3D->dataProvider();
+
+  QGSCOMPARENEAR( prLines3D->sourceExtent().xMinimum(), 0.0, 0.001 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent().xMaximum(), 322355.7101, 0.01 );
```

Comment:
Can you be consistent in the test? Using only 0.01 is fine to me.

Reply:
fixed!